src/deluge/model/clip/clip.cpp

```diff
@@ -941,10 +943,45 @@ int32_t Clip::solicitParamManager(Song* song, ParamManager* newParamManager, Cli
 }
 
 void Clip::clear(Action* action, ModelStackWithTimelineCounter* modelStack) {
+	//New community feature as part of Automation Clip View Implementation
+	//If this is enabled, then when you are in a regular Instrument Clip View (Synth, Kit, MIDI, CV), clearing a clip
+	//will only clear the Notes and MPE data (NON MPE automations remain intact).
+
+	//If this is enabled, if you want to clear NON MPE automations, you will enter Automation Clip View and clear the clip there.
+
+	ModelStackWithThreeMainThings* modelStackWithThreeMainThings =
+	    modelStack->addOtherTwoThingsButNoNoteRow(output->toModControllable(), &paramManager);
+
 	if (paramManager.containsAnyParamCollectionsIncludingExpression()) {
-		ModelStackWithThreeMainThings* modelStackWithThreeMainThings =
-		    modelStack->addOtherTwoThingsButNoNoteRow(output->toModControllable(), &paramManager);
-		paramManager.deleteAllAutomation(action, modelStackWithThreeMainThings);
+		ParamCollectionSummary* summary = paramManager.summaries;
+
+		int32_t i = 0;
+
+		while (summary->paramCollection) {
```

Comment:
Checking for true is likely to lead to bugs down the road (an uninitialized summary will pass this check and then crash). Can this be rewritten as a for loop with a concrete length?

Reply:
Yes let me look at this.